src/prommis/hydrogen_decrepitation/hydrogen_decrepitation_furnace.py

```diff
@@ -0,0 +1,1651 @@
+
+        self.relative_thickness_ratio_insulation_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=1,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Relative thickness ratio of insulation material 2",
+        )
+
+        self.thickness_metal_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=4 * 1e-3,
+            units=pyunits.m,
+            bounds=(0, None),
+            doc="Thickness of metal material 2",
+        )
+
+        self.relative_thickness_ratio_metal_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=0.01,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Relative thickness ratio of metal material 2",
+        )
+
+        self.total_heat_duty = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=3500,
+            units=pyunits.W,
+            bounds=(0, None),
+            doc="Total heat duty required for decrepitation",
+        )
+
+    def _make_geometry(self):
+        """This section contains equations related to the geometry of the reactor model."""
+
+        # feed properties
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="volumetric flow of solid feed",
+        )
+        def flow_vol_feed_eqn(b, t):
+            return b.flow_vol_feed[t] == pyunits.convert(
+                b.solid_in[t].flow_mass / b.sample_density,
+                to_units=pyunits.m**3 / pyunits.s,
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="mass of solid feed",
+        )
+        def sample_mass_eqn(b, t):
+            return b.sample_mass[t] == pyunits.convert(
+                b.solid_in[t].flow_mass * b.decrepitation_duration, to_units=pyunits.kg
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="volume of solid feed",
+        )
+        def sample_volume_eqn(b, t):
+            return b.sample_volume[t] == pyunits.convert(
+                b.flow_vol_feed[t] * b.decrepitation_duration, to_units=pyunits.m**3
+            )
+
+        # furnace chamber dimensions
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="internal volume of the furnace",
+        )
+        def furnace_chamber_volume_eqn(b, t):
+            return b.furnace_chamber_volume[t] == pyunits.convert(
+                b.chamber_to_sample_ratio * b.sample_volume[t], to_units=pyunits.m**3
+            )
+
+        @self.Expression(
+            self.flowsheet().config.time, doc="Radius of the furnace chamber"
+        )
+        def radius_chamber(b, t):
+            return (b.furnace_chamber_volume[t] / (6 * Constants.pi)) ** (1 / 3)
```

Comment:
What's the shape of the chamber here? This isn't the formula I'd expect for a sphere or hemisphere.